### PR TITLE
RPC for requesting objects, schema & version

### PR DESCRIPTION
**Pull request by tribal-tec**
- v1/ prefix for endpoints is gone
- set-<object> replaces <object> notification to update objects
- added get-<object> request to retrieve current object state

### CODE REVIEW

**Comment by tribal-tec** [comment]
I'm waiting for @rolandjitsu to test this, plus I should bump the version now after breaking the API...

File: plugins/RocketsPlugin/RocketsPlugin.cpp [hunk 1]
```diff
@@ -47,7 +47,6 @@
 
 namespace
 {
-const std::string ENDPOINT_API_VERSION = "v1/";
 const std::string ENDPOINT_ANIMATION_PARAMS = "animation-parameters";
```
Comment (rolandjitsu) [inline plugins/RocketsPlugin/RocketsPlugin.cpp]: Let's group these methods/endpoints according to which can be a request or a notification. Maybe also according to which have `set-*`/`get-*`.

Comment (tribal-tec) [reply plugins/RocketsPlugin/RocketsPlugin.cpp]: Done

File: plugins/RocketsPlugin/RocketsPlugin.cpp [hunk 2]
```diff
@@ -47,36 +47,47 @@
...
-const std::string METHOD_CHUNK = "chunk";
+const std::string METHOD_SNAPSHOT = "snapshot";
+// METHOD_REQUEST_MODEL_UPLOAD from BinaryRequests.h
+
+// JSONRPC synchronous requests
 const std::string METHOD_GET_INSTANCES = "get-instances";
+const std::string METHOD_IMAGE_JPEG = "image-jpeg";
 const std::string METHOD_INSPECT = "inspect";
-const std::string METHOD_QUIT = "quit";
 const std::string METHOD_REMOVE_MODEL = "remove-model";
```
Comment (rolandjitsu) [inline plugins/RocketsPlugin/RocketsPlugin.cpp]: Aren't `'remove-model'` and `'update-model'` jsonrpc notifications?

File: plugins/RocketsPlugin/RocketsPlugin.cpp [hunk 3]
```diff
@@ -47,36 +47,47 @@
...
 const std::string ENDPOINT_SCENE = "scene";
 const std::string ENDPOINT_SCENE_PARAMS = "scene-parameters";
-const std::string ENDPOINT_SIMULATION_HISTOGRAM = "simulation-histogram";
-const std::string ENDPOINT_STATISTICS = "statistics";
 const std::string ENDPOINT_STREAM = "stream";
-const std::string ENDPOINT_STREAM_TO = "stream-to";
+const std::string ENDPOINT_VOLUME_PARAMS = "volume-parameters";
+
+// REST GET, JSONRPC get-* request
+const std::string ENDPOINT_STATISTICS = "statistics";
```
Comment (rolandjitsu) [inline plugins/RocketsPlugin/RocketsPlugin.cpp]: So `'statistics'` and `'version'` can now be fetched using a jsonrpc request with `{method: 'get-statistics'}`/`{method: 'get-version'}`?

How about when statistics is broadcasted from the server? Is it broadcasted as `{method: 'get-statistics'}` or `{method: 'statistics'}`?

Comment (tribal-tec) [reply plugins/RocketsPlugin/RocketsPlugin.cpp]: Yes, they can be requested as you described.

The broadcast haven't changed, so it's ```statistics``` still

File: plugins/RocketsPlugin/RocketsPlugin.cpp (hunk 2 above)
Comment (tribal-tec) [reply plugins/RocketsPlugin/RocketsPlugin.cpp]: You can use them with notifications if you want. But they can return true/false on success/error.

File: plugins/RocketsPlugin/RocketsPlugin.cpp [hunk 4]
```diff
@@ -739,15 +747,27 @@ class RocketsPlugin::Impl
...
+    void _handleSchemaRPC()
+    {
+        RpcDocumentation doc{"Get the schema of the given endpoint", "endpoint",
+                             "name of the endpoint to get its schema"};
+        _handleRPC<SchemaParam, std::string>(
+            METHOD_SCHEMA, doc, [& schemas = _schemas](const auto& param) {
+                if (schemas.count(param.endpoint) == 0)
+                    throw rockets::jsonrpc::response_error("Endpoint not found",
+                                                           -12347);
+                return schemas[param.endpoint];
```
Comment (tribal-tec) [inline plugins/RocketsPlugin/RocketsPlugin.cpp]: json.parse missing here

**Pull request merged**